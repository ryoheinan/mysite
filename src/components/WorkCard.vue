<template>
  <div class="Box px-3 mb-3 pt-5 pb-1">
    <div class="flex-auto">
      <h3>{{ workData.title }}</h3>
      <p class="mb-1">
        Powered by <span class="strong">{{ workData.technology }}</span>
      </p>
      <p class="text-gray-light">{{ workData.description }}</p>

      <span
        v-if="workData.category == 'Hobby'"
        title="Label: Hobby"
        class="Label Label--success v-align-middle"
        >Hobby</span
      >
      <span
        v-else-if="workData.category == 'School'"
        title="Label: School"
        class="Label mt-1 Label--info v-align-middle"
        >School</span
      >
      <span
        v-else-if="workData.category == 'Work'"
        title="Label: Work"
        class="Label mt-1 Label--warning v-align-middle"
        >Work</span
      >
    </div>
    <div class="mt-3 text-right">
      <a
        v-for="urlItem in workData.urls"
        :key="urlItem.text"
        :href="urlItem.url"
        class="btn ml-2 mb-2"
        role="button"
        target="_blank"
        rel="noopener noreferrer nofollow"
        >{{ urlItem.text }}</a
      >
    </div>
  </div>
</template>

<script lang="ts">
  import { defineComponent, PropType } from 'vue'
  import { Work } from '../module/workconfig'

  export default defineComponent({
    name: 'WorkCard',
    props: {
      textLeft: {
        type: Boolean,
        default: false,
        required: false,
      },
      workData: {
        type: Object as PropType<Work>,
        required: true,
      },
    },
  })
</script>

<style lang="scss" scoped>
  .strong {
    font-size: 1.1rem;
    font-weight: 500;
  }
</style>
