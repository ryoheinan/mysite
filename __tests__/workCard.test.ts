import { mount } from '@vue/test-utils'
import WorkCard from '../src/components/WorkCard.vue'
import { describe, expect, it } from 'vitest'

describe.each([['Hobby'], ['School'], ['Work']])(
  'WorkCard component (category: %s)',
  (categoryName) => {
    expect(WorkCard).toBeTruthy()

    const wrapper = mount(WorkCard, {
      props: {
        workData: {
          title: 'すごい成果物',
          description: 'これは\n説明です。',
          technology: 'Vue 3, TypeScript, Django Rest Framework, PostgreSQL',
          category: categoryName as 'School' | 'Hobby' | 'Work',
          year: '2021 夏 ~  2022 初頭',
          image: {
            src: 'work.jpg',
            alt: 'workのスクリーンショット画像',
            srcset: 'work.webp',
          },
          urls: [
            {
              text: 'Source code (client)',
              url: 'https://example.com/client',
            },
            {
              text: 'Source code (backend)',
              url: 'https://example.com/backend',
            },
          ],
        },
      },
    })

    it('titleが正しく設定されているか', () => {
      expect(wrapper.find('h3').text()).toBe('すごい成果物')
    })
    it('descriptionが正しく設定されているか', () => {
      expect(wrapper.find('[data-test="description"]').text()).toEqual(
        'これは\n説明です。'
      )
    })
    it('technologyが正しく設定されているか', () => {
      expect(wrapper.find('[data-test="technology"]').text()).toBe(
        'Vue 3, TypeScript, Django Rest Framework, PostgreSQL'
      )
    })
    it(`category (${categoryName}) が正しく設定されているか`, () => {
      expect(wrapper.text()).toContain(categoryName)
    })
    it('yearが正しく設定されているか', () => {
      expect(wrapper.find('[data-test="year"]').text()).toBe(
        '2021 夏 ~  2022 初頭'
      )
    })
    it('画像が正しく設定されているか', () => {
      expect(wrapper.find('img').attributes().src).toBe('/img/work.jpg')
      expect(wrapper.find('img').attributes().alt).toBe(
        'workのスクリーンショット画像'
      )
      expect(wrapper.find('source').attributes().srcset).toBe('/img/work.webp')
    })
    it('urlが正しく設定されているか', () => {
      const buttons = wrapper.findAll('a')
      expect(buttons).toHaveLength(2)
      expect(buttons[0].attributes().href).toBe('https://example.com/client')
      expect(buttons[1].attributes().href).toBe('https://example.com/backend')
      expect(buttons[0].text()).toBe('Source code (client)')
      expect(buttons[1].text()).toBe('Source code (backend)')
    })
    it('Snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  }
)
