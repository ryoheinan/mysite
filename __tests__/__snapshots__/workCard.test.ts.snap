// Vitest Snapshot v1

exports[`WorkCard component (category: Hobby) > Snapshot 1`] = `
"<div class=\\"Box px-md-3 px-2 mb-3 pt-5 pb-1\\" data-v-0f391d71=\\"\\">
  <h3 class=\\"mb-2\\" data-v-0f391d71=\\"\\">すごい成果物</h3>
  <div class=\\"container-lg clearfix\\" data-v-0f391d71=\\"\\">
    <div class=\\"col-md-6 float-left mb-2 mb-md-0\\" data-v-0f391d71=\\"\\">
      <p class=\\"mb-1\\" data-v-0f391d71=\\"\\"> Powered by <span class=\\"strong\\" data-test=\\"technology\\" data-v-0f391d71=\\"\\">Vue 3, TypeScript, Django Rest Framework, PostgreSQL</span></p>
      <p class=\\"mb-3\\" data-test=\\"year\\" data-v-0f391d71=\\"\\">2021 夏 ~ 2022 初頭</p>
      <p class=\\"mb-2 text-gray-light description\\" data-test=\\"description\\" data-v-0f391d71=\\"\\">これは
        説明です。</p><span title=\\"Label: Hobby\\" class=\\"Label Label--success v-align-middle\\" data-v-0f391d71=\\"\\">Hobby</span>
    </div>
    <div class=\\"col-md-6 float-md-left\\" data-v-0f391d71=\\"\\">
      <div class=\\"img text-center\\" data-v-0f391d71=\\"\\">
        <picture data-v-0f391d71=\\"\\">
          <source type=\\"image/webp\\" srcset=\\"/img/work.webp\\" data-v-0f391d71=\\"\\"><img class=\\"img\\" alt=\\"workのスクリーンショット画像\\" src=\\"/img/work.jpg\\" data-v-0f391d71=\\"\\">
        </picture>
      </div>
    </div>
  </div>
  <div class=\\"mt-3\\" data-v-0f391d71=\\"\\"><a href=\\"https://example.com/client\\" class=\\"btn mr-2 mb-2\\" role=\\"button\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-v-0f391d71=\\"\\">Source code (client)</a><a href=\\"https://example.com/backend\\" class=\\"btn mr-2 mb-2\\" role=\\"button\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-v-0f391d71=\\"\\">Source code (backend)</a></div>
</div>"
`;

exports[`WorkCard component (category: School) > Snapshot 1`] = `
"<div class=\\"Box px-md-3 px-2 mb-3 pt-5 pb-1\\" data-v-0f391d71=\\"\\">
  <h3 class=\\"mb-2\\" data-v-0f391d71=\\"\\">すごい成果物</h3>
  <div class=\\"container-lg clearfix\\" data-v-0f391d71=\\"\\">
    <div class=\\"col-md-6 float-left mb-2 mb-md-0\\" data-v-0f391d71=\\"\\">
      <p class=\\"mb-1\\" data-v-0f391d71=\\"\\"> Powered by <span class=\\"strong\\" data-test=\\"technology\\" data-v-0f391d71=\\"\\">Vue 3, TypeScript, Django Rest Framework, PostgreSQL</span></p>
      <p class=\\"mb-3\\" data-test=\\"year\\" data-v-0f391d71=\\"\\">2021 夏 ~ 2022 初頭</p>
      <p class=\\"mb-2 text-gray-light description\\" data-test=\\"description\\" data-v-0f391d71=\\"\\">これは
        説明です。</p><span title=\\"Label: School\\" class=\\"Label mt-1 Label--info v-align-middle\\" data-v-0f391d71=\\"\\">School</span>
    </div>
    <div class=\\"col-md-6 float-md-left\\" data-v-0f391d71=\\"\\">
      <div class=\\"img text-center\\" data-v-0f391d71=\\"\\">
        <picture data-v-0f391d71=\\"\\">
          <source type=\\"image/webp\\" srcset=\\"/img/work.webp\\" data-v-0f391d71=\\"\\"><img class=\\"img\\" alt=\\"workのスクリーンショット画像\\" src=\\"/img/work.jpg\\" data-v-0f391d71=\\"\\">
        </picture>
      </div>
    </div>
  </div>
  <div class=\\"mt-3\\" data-v-0f391d71=\\"\\"><a href=\\"https://example.com/client\\" class=\\"btn mr-2 mb-2\\" role=\\"button\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-v-0f391d71=\\"\\">Source code (client)</a><a href=\\"https://example.com/backend\\" class=\\"btn mr-2 mb-2\\" role=\\"button\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-v-0f391d71=\\"\\">Source code (backend)</a></div>
</div>"
`;

exports[`WorkCard component (category: Work) > Snapshot 1`] = `
"<div class=\\"Box px-md-3 px-2 mb-3 pt-5 pb-1\\" data-v-0f391d71=\\"\\">
  <h3 class=\\"mb-2\\" data-v-0f391d71=\\"\\">すごい成果物</h3>
  <div class=\\"container-lg clearfix\\" data-v-0f391d71=\\"\\">
    <div class=\\"col-md-6 float-left mb-2 mb-md-0\\" data-v-0f391d71=\\"\\">
      <p class=\\"mb-1\\" data-v-0f391d71=\\"\\"> Powered by <span class=\\"strong\\" data-test=\\"technology\\" data-v-0f391d71=\\"\\">Vue 3, TypeScript, Django Rest Framework, PostgreSQL</span></p>
      <p class=\\"mb-3\\" data-test=\\"year\\" data-v-0f391d71=\\"\\">2021 夏 ~ 2022 初頭</p>
      <p class=\\"mb-2 text-gray-light description\\" data-test=\\"description\\" data-v-0f391d71=\\"\\">これは
        説明です。</p><span title=\\"Label: Work\\" class=\\"Label mt-1 Label--warning v-align-middle\\" data-v-0f391d71=\\"\\">Work</span>
    </div>
    <div class=\\"col-md-6 float-md-left\\" data-v-0f391d71=\\"\\">
      <div class=\\"img text-center\\" data-v-0f391d71=\\"\\">
        <picture data-v-0f391d71=\\"\\">
          <source type=\\"image/webp\\" srcset=\\"/img/work.webp\\" data-v-0f391d71=\\"\\"><img class=\\"img\\" alt=\\"workのスクリーンショット画像\\" src=\\"/img/work.jpg\\" data-v-0f391d71=\\"\\">
        </picture>
      </div>
    </div>
  </div>
  <div class=\\"mt-3\\" data-v-0f391d71=\\"\\"><a href=\\"https://example.com/client\\" class=\\"btn mr-2 mb-2\\" role=\\"button\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-v-0f391d71=\\"\\">Source code (client)</a><a href=\\"https://example.com/backend\\" class=\\"btn mr-2 mb-2\\" role=\\"button\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" data-v-0f391d71=\\"\\">Source code (backend)</a></div>
</div>"
`;
